test/e2e/operatorhub_test.go

```diff
@@ -67,6 +67,30 @@ var _ = Describe("operatorhub", func() {
 				return nil
 			}, defaultTimeout, 3).Should(BeNil())
 
+			By("ensuring the operatorHub status reflects that the default catalogsources have been disabled successfully")
+			Eventually(func() error {
+				oh := &configv1.OperatorHub{}
+				if err := k8sClient.Get(ctx, nn, oh); err != nil {
+					return err
+				}
+				defaultSources := map[string]struct{}{
```

Comment:
Instead of hard coding the defaults we should probably use something like [GetGlobals()](https://github.com/operator-framework/operator-marketplace/blob/master/pkg/defaults/defaults.go#L87)

Reply:
I disagree here for two reasons:
1. If the `getDefaults` function ever breaks, this test would pass despite not having the expected status as the length of the array would be 0. We could add a `len > 0` check, but we still wouldn't be guaranteed that the status of each defaultCatalogSource exists unless we check that `len == # defaultCatalogSources`. At that point, this code needs to be updated anyways, so I think its better to put the onus on the individual introducing the new default to update this function.
2. Even if the `getDefaults` is tested elsewhere and never breaks, the status of this CR clearly isn't tested in the existing suite, otherwise this issue would have been caught earlier. As such, I want stronger testing in this space.

Reply:
Which `getDefaults` are you talking about here exactly? 

Once you

```
err := PopulateGlobals()
Expect(err).ToNot(HaveOccurred())
GetGlobals()
```
Ref: [PopulateGlobals()](https://github.com/operator-framework/operator-marketplace/blob/master/pkg/defaults/defaults.go#L111)

I don't see how the length of the array can be 0. 

> As such, I want stronger testing in this space.

Exactly my reasoning for not hard coding the name/number of default CatalogSources. Consider this: tomorrow if someone else adds two new default sources, she'll be unaware that this test hard codes the defaults, and therefore this test will no longer be strong.

Reply:
Sorry for the late response. This test would prevent the developers changes from coming in. Alternatively, if a developer accidentally removed one or all of the default catalogSources, there would be no signal if the tests relied on the PopulateGlobals function.